src/index.ts

```diff
@@ -265,12 +265,21 @@ export class Paginator {
     const callback = parsedArguments.callback!;
     if (parsedArguments.autoPaginate) {
       const results = new Array<{}>();
-      paginator.runAsStream_(parsedArguments, originalMethod)
-          .on('error', callback)
-          .on('data', (data: {}) => results.push(data))
-          .on('end', () => callback(null, results));
+      const promise = new Promise((resolve, reject) => {
+        paginator.runAsStream_(parsedArguments, originalMethod)
+            .on('error', reject)
+            .on('data', (data: {}) => results.push(data))
+            .on('end', () => resolve([results]));
```

Comment:
Another nit I have is that we're packaging this up to match the expected return value of a Promise before we even know what the user wants. I think it would be slightly more efficient if we just packed is up via promise chain

```js
return promise.then(results => [results]);
```

Reply:
done, thanks